### DISCUSSION TITLE
meow-block select wrong region in org-mode source block

### DISCUSSION THREAD

**Discussion by K-gihu** [meow-edit/meow#405] [Q&A] [answered]
As the title says, if you write the following text in org-mode, then put the cursor before "square" and evaluate `meow-block`, it will mark the region between the square parenthesis and the round parenthesis:
```org-mode
#+begin_src elisp
(setq meow-char-thing-table
      '(
        (?\[ . square)
        ))
#+end_src
```

Actually in elisp-mode, there is no such problem, but in org-mode things are different.
I wonder if there is a way to fix this problem ?

**Comment by DogLooksGood** [comment, marked as answer]
Because in Elisp mode, the code is parsed as Elisp, so `?` is treated as a prefix. However in org-mode, there's no syntax for Elisp. 
Meow uses the builtin syntax pass to detect the levels of paired parentheses, so there's no easy way to fix. 

I usually use `C-c '` to narrow to the code snippet for editing.

**Reply by K-gihu** [reply]
Thank you for the reply.
